src/evergreen/api.py

```diff
@@ -544,12 +538,24 @@ def tasks_by_project(self, project_id: str, statuses: Optional[List[str]] = None
         :param statuses: the types of statuses to get tasks for.
         :return: The list of matching tasks.
         """
-        url = self._create_url(
-            "/projects/{project_id}/versions/tasks".format(project_id=project_id)
-        )
+        url = self._create_url(f"/projects/{project_id}/versions/tasks")
         params = {"status": statuses} if statuses else None
         return [Task(json, self) for json in self._paginate(url, params)]  # type: ignore[arg-type]
 
+    def tasks_by_project_and_commit(
+        self, project_id: str, commit_hash: str, params: Dict = None
```

Comment:
[nit] Since `params` can be `None`, I think it should be an `Optional[Dict]`

Reply:
nice catch, thanks!